vllm/entrypoints/openai/serving_completion.py

```diff
@@ -150,15 +151,16 @@ async def create_completion(
                     log_tracing_disabled_warning()
 
                 if isinstance(sampling_params, BeamSearchParams):
-                    if not isinstance(self.engine_client, AsyncLLMEngine):
-                        raise ValueError(
-                            "Beam search in the API server is only supported"
-                            " with AsyncLLMEngine. please add "
-                            "`--disable-frontend-multiprocessing` to "
-                            "use beam search.")
+                    assert isinstance(self.engine_client,
+                                    (AsyncLLMEngine,
+                                    MQLLMEngineClient)), \
+                    "Beam search is only supported with" \
+                    "AsyncLLMEngine and MQLLMEngineClient."
                     generator = self.engine_client.beam_search(
-                        prompt_inputs["prompt_token_ids"], request_id_item,
-                        sampling_params)
+                        prompt_inputs["prompt_token_ids"],
+                        request_id,
```

Comment:
what's the difference between `request_id` and `request_id_item` ?

Reply:
Ahh yeah it's supposed to be `request_id_item`, I was a bit careless when I was doing some refactoring. I think completion supports multiple prompts so each prompt has it's own `request_id_item` for a general `request_id`.